src/pybind/mgr/cephadm/module.py

```diff
@@ -1692,12 +1694,13 @@ def remove_daemons(self, names):
     def remove_service(self, service_name: str) -> str:
         self.log.info('Remove service %s' % service_name)
         self._trigger_preview_refresh(service_name=service_name)
+        if service_name in self.spec_store:
+            if self.spec_store[service_name].spec.service_type in ('mon', 'mgr'):
+                return f'Unable to remove {service_name} service.\n' \
+                       f'Note, you might want to mark the {service_name} service as "unmanaged"'
```

Comment:
Since this is only ever printed for mgr and mon, is it a good idea to tell users to mark those services as unmanaged? If they do and then proceed to remove them won't everything break down?

Reply:
Right. have a better idea for a text? 

I mean if we decide that we **always** want to have specs for mon and mgr, then we simply should not allow users to delete those services. period. Cause otherwise we can no longer manage the cluster. But should we really recommend setting unmanaged?